vertx-web/src/main/java/io/vertx/ext/web/handler/impl/OAuth2AuthHandlerImpl.java

```diff
@@ -145,6 +149,10 @@ private String authURI(String redirectURL, String state) {
       config.mergeIn(extraParams);
     }
 
+    if (scopes.size() > 0) {
```

Comment:
Maybe we could ignore this check and always add scopes. This would avoid bad request errors on google for example